test/Main.purs

```diff
@@ -179,87 +159,164 @@ main = do
     "./../foo/../"
 
   test' "canonicalize - 1 down, 1 up"
-    (canonicalize $ parentDir $ dirFoo)
+    (canonicalize $ parentOf $ dirFoo)
     "./"
 
   test' "canonicalize - 2 down, 2 up"
-    (canonicalize (parentDir (parentDir (dirFoo </> dirBar))))
+    (canonicalize (parentOf (parentOf (dirFoo </> dirBar))))
     "./"
 
-  test "renameFile - single level deep"
-    (renameFile' dropExtension (file (reflectNonEmpty $ SProxy :: SProxy "image.png")))
+  test' "canonicalize - 2 up from root"
+    (canonicalize (parentOf (parentOf rootDir)))
+    "/"
+
+  test "canonicalize /foo/../bar/ = /bar"
+    (canonicalize (rootDir </> dirFoo <..> dirBar))
+    (rootDir </> dirBar)
+
+  test "relativeTo rootDir rootDir = currentDir"
+    (relativeTo rootDir rootDir)
+    (currentDir)
+
+  test' "(rootDir </> dirFoo) `relativeTo` (rootDir </> dirFoo) = ./"
+    ((rootDir </> dirFoo) `relativeTo` (rootDir </> dirFoo))
+    "./"
+
+  test' "(rootDir </> dirFoo) `relativeTo` rootDir = currentDir </> dirFoo"
+    ((rootDir </> dirFoo) `relativeTo` rootDir)
+    "./foo/"
+
+  test' "(rootDir </> dirFoo) `relativeTo` (rootDir </> dirBar) = currentDir <..> dirFoo"
```

Comment:
What you think about having `currentDir <..> _` as separate function? might it be useful?

Reply:
That's `parentOf`.